library/libGUI/lib_window_akros_main_menu.ks

```diff
@@ -0,0 +1,167 @@
+		if process_needs_redraw(process_state){
+			draw_window_akros_main_menu(process_state).
+		}
+
+		local wnd is get_process_window(process_state).
+		local last_abort is process_state[2].
+
+		if abort<>last_abort{
+			draw_outline(wnd).
+			set process_state[1] to "program_selection".
+			local child_process is open_window_menu(
+				"Select program:",
+				list("Vessel stats","Back","Quit akrOS"),
+				wnd
+			).
+			set process_state[3] to child_process.
+		}
+	}
+	else if run_mode="program_selection"{
+		local child_process is process_state[3].
+		if process_needs_redraw(process_state){ // pass redraw event to child
+			invalidate_process_window(child_process).
+			set process_state[0][3] to false. // redraw no longer needed
+		}
+		local selection is update_process(child_process).
+		if process_finished(child_process){
+			draw_outline(wnd).
+			if selection="Quit akrOS"{
+				local all_proc is process_state[4].
+				local i is 0.
+				until i=all_proc:length{
+					end_process(all_proc[i]).
+					set i to i+1.
+				}
+				return 0.
+			}
+			else if selection="Back"{
+				set process_state[1] to "title_screen".
+				invalidate_process_window(process_state).
+			}
+			else{
+				local len is process_state[5]:length.
+				local lw is list().
+				local i is 0.
+				until i=len{
+					lw:add(i).
+					set i to i+1.
+				}
+				set child_process to open_window_menu(
+					"Select window",lw,wnd
+				).
+				set process_state[1] to "window_selection".
+				set process_state[3] to child_process.
+				set process_state[6] to selection.
+			}
+		}
+	}
+	else if run_mode="window_selection"{
+		local child_process is process_state[3].
+		if process_needs_redraw(process_state){ // pass redraw event to child
+			invalidate_process_window(child_process).
+			set process_state[0][3] to false. // redraw no longer needed
+		}
+		local selection is update_process(child_process).
+		if process_finished(child_process){
+			draw_outline(wnd).
+			
+			local other_process is get_process_from_name(
+				process_state[6],process_state[5],process_state[4],selection
+			).
+
+			if selection<>0{ // menu is still there
+				local all_proc is process_state[4].
+				all_proc:add(other_process).
+				invalidate_process_window(process_state).
+				set process_state[1] to "title_screen".
+			}
+			else{ // menu must disappear to show program
+				set process_state[3] to other_process.
+				set process_state[1] to "waiting_for_foreground".
+			}
+		}
+	}
+	else if run_mode="waiting_for_foreground"{
+		local child_process is process_state[3].
+		if process_needs_redraw(process_state){ // pass redraw event to child
+			invalidate_process_window(child_process).
+			set process_state[0][3] to false. // redraw no longer needed
+		}
+		update_process(child_process).
+		if process_finished(child_process){
+			draw_outline(wnd).
+			invalidate_process_window(process_state).
+			set process_state[1] to "title_screen".
+		}
+	}
+	
+
+	set process_state[2] to abort.
+	return -1.
```

Comment:
It looks like you haven't read kOS crash logs too. kOS dumps it's opcode cache to KSP_Data/output_log.txt.
http://ksp-kos.github.io/KOS_DOC/language/user_functions.html#inconsistent-returns
I see:

> The internal stack will not get corrupted by this error, as some experienced programmers might expect upon hearing this (because secretly all kerboscript user functions return a value even if it’s never used, so there’s universally always something to pop off the stack even for the empty return statements.)